beacon-chain/sync/service.go

```diff
@@ -158,22 +193,34 @@ func (ss *Service) run() {
 				log.Error("Received malformed beacon block p2p message")
 				continue
 			}
+
 			block := types.NewBlock(response.Block)
-			h, err := block.Hash()
+			blockHash, err := block.Hash()
 			if err != nil {
 				log.Errorf("Could not hash received block: %v", err)
 			}
-			blockExists, err := ss.chainService.ContainsBlock(h)
+
+			blockExists, err := ss.chainService.ContainsBlock(blockHash)
 			if err != nil {
 				log.Errorf("Can not check for block in DB: %v", err)
 				continue
 			}
 			if blockExists {
 				continue
 			}
-			log.WithField("blockHash", fmt.Sprintf("0x%x", h)).Debug("Sending newly received block to subscribers")
-			// We send out a message over a feed.
+
+			// Verify attestation coming from proposer then forward it to the subscribers.
+			if response.Attestation != nil {
+				attestation := types.NewAttestation(response.Attestation)
+				if err := ss.verifyAttestation(block.ParentHash(), attestation); err != nil {
+					log.WithField("attestationHash", fmt.Sprintf("0x%x", attestation.Key())).Debug("Sending newly received attestation to subscribers")
+					ss.chainService.IncomingAttestationFeed().Send(attestation)
```

Comment:
Hmm, where is this attestation going? There's no need to send it anywhere, its only used to validate the block.

Reply:
I could be wrong here. My thought is, if we don't send it anywhere, who is going to include it to the block? Right now it's sending it to the Block Chain service then relay it to RPC then to Proposer to include for the next beacon block.

Reply:
That's a good point. I assumed that the proposer would broadcast the attestation through the normal attestation channel. This might be better since it reduces an extra roundtrip. I'd leave this out though since we don't know how attestation aggregation is going to work yet.